detection_rules/packaging.py

```diff
@@ -74,77 +74,54 @@ def load_versions(current_versions: dict = None):
-        }
+    echo = click.echo if verbose else (lambda x: None)
+
+    already_deprecated = set(rule_deprecations)
+    deprecated_rules = set(rule.id for rule in rules if rule.contents.metadata.maturity == "deprecated")
+    new_rules = set(rule.id for rule in rules if rule.contents.latest_version is None) - deprecated_rules
+    changed_rules = set(rule.id for rule in rules if rule.contents.is_dirty) - deprecated_rules
 
-    new_rules = [rule for rule in rules if rule.contents.latest_version is None]
-    changed_rules = [rule for rule in rules if rule.contents.is_dirty]
     # manage deprecated rules
-    newly_deprecated = []
-    rule_deprecations = {}
-
-    if deprecated_rules:
-        rule_deprecations = load_etc_dump('deprecated_rules.json')
-
-        for rule in deprecated_rules:
-            if rule.id not in rule_deprecations:
-                rule_deprecations[rule.id] = {
-                    'rule_name': rule.name,
-                    'deprecation_date': rule.contents.metadata.deprecation_date,
-                    'stack_version': current_stack_version() + ".0"
-                }
-                newly_deprecated.append(rule.id)
-
-    # update the document with the new rules
-    # if current_versions != new_versions???
-    if new_rules or changed_rules or newly_deprecated:
-        if verbose:
-            click.echo('Rule hash changes detected!')
+    newly_deprecated = deprecated_rules - already_deprecated
 
-        if save_changes:
-            if changed_rules or (new_rules and add_new):
-                if add_new:
-                    for rule in new_rules:
-                        current_versions[rule.id] = rule.contents.lock_info()
-                for rule in changed_rules:
-                    current_versions[rule.id] = rule.contents.lock_info()
+    if not (new_rules or changed_rules or newly_deprecated):
+        return changed_rules, list(new_rules), newly_deprecated
 
-                current_versions = OrderedDict(sorted(current_versions.items(), key=lambda x: x[1]['rule_name']))
+    echo('Rule changes detected!')
 
-                save_etc_dump(current_versions, 'version.lock.json')
+    if not save_changes:
+        echo('run `build-release --update-version-lock` to update version.lock.json and deprecated_rules.json')
+        return changed_rules, list(new_rules), newly_deprecated
 
-                if verbose:
-                    click.echo('Updated version.lock.json file')
+    for rule in rules:
+        contents = rule.contents.lock_info(bump=not exclude_version_update)
 
-            if newly_deprecated:
-                save_etc_dump(OrderedDict(sorted(rule_deprecations.items(), key=lambda e: e[1]['rule_name'])),
-                              'deprecated_rules.json')
+        if rule.contents.metadata.maturity == "production":
+            current_versions[rule.id] = contents
+        elif rule.id in newly_deprecated:
+            rule_deprecations[rule.id] = contents
```

Comment:
I can't remember if we were actually locking versions at deprecation time too (before the refactor)

Reply:
it was hard to tell, to be honest

Reply:
the contents of the deprecation file is different from the version lock

```json
  "139c7458-566a-410c-a5cd-f80238d6a5cd": {
    "deprecation_date": "2021/04/15",
    "rule_name": "SQL Traffic to the Internet",
    "stack_version": "7.14.0"
```

Reply:
94b1c2e5e7b3df792f2f12245737db1d5e4d5a6b

Reply:
94b1c2e5e7b3df792f2f12245737db1d5e4d5a6b

Reply:
```suggestion
```